arch/x86/adler32_avx.c

```diff
@@ -19,7 +19,14 @@ static inline uint64_t hsum(__m256i x) {
     __m256i sum1 = _mm256_shuffle_epi32(x, 0x0E);
     __m256i sum2 = _mm256_add_epi64(x, sum1);
     __m128i sum3 = _mm256_extracti128_si256(sum2, 1);
+#ifdef __X86_64__
     return _mm_cvtsi128_si64(_mm_add_epi64(_mm256_castsi256_si128(sum2), sum3));
+#else
+    __m128i retVec = _mm_add_epi64(_mm256_castsi256_si128(sum2), sum3);
```

Comment:
Don't forget `retVec` and `retVal` are not in the proper casing for the project.